pyomo/contrib/sensitivity_toolbox/tests/test_sens_unit.py

```diff
@@ -0,0 +1,807 @@
+
+        # Here we rely on the order of paramConst
+        for param, var, con in zip(param_list, var_list, param_const.values()):
+            self.assertEqual(con.body.to_string(), (var - param).to_string())
+
+    def test_param_const_indexed(self):
+        model = make_indexed_model()
+        param_list = [model.eta]
+        sens = SensitivityInterface(model, clone_model=False)
+        sens.setup_sensitivity(param_list)
+
+        block = sens.block
+        param_const = block.paramConst
+
+        param_var_map = ComponentMap((param, var)
+                for var, param, _, _ in block._sens_data_list)
+
+        for con in param_const.values():
+            var_list = list(identify_variables(con.expr))
+            mut_param_list = list(identify_mutable_parameters(con.expr))
+            self.assertEqual(len(var_list), 1)
+            self.assertEqual(len(mut_param_list), 1)
+            self.assertIs(var_list[0], param_var_map[mut_param_list[0]])
+            self.assertEqual(con.body.to_string(),
+                    (var_list[0]-mut_param_list[0]).to_string())
+
+    def test_param_const_vars(self):
+        model = make_indexed_model()
+        model.x.fix()
+        var_list = [model.x[2], model.x[1]]
+        sens = SensitivityInterface(model, clone_model=False)
+        sens.setup_sensitivity(var_list)
+
+        block = sens.block
+        param_const = block.paramConst
+        self.assertEqual(len(var_list), len(block.paramConst))
+
+        var_param_map = ComponentMap((var, param)
+                for var, param, _, _ in block._sens_data_list)
+        param_list = [var_param_map[var] for var in var_list]
+
+        # Here we rely on the order of paramConst
+        for param, var, con in zip(param_list, var_list, param_const.values()):
+            self.assertEqual(con.body.to_string(), (var - param).to_string())
+
+    def test_suffixes_setup(self):
+        model = make_indexed_model()
+        param_list = [model.eta[2], model.eta[1]]
+        sens = SensitivityInterface(model, clone_model=False)
+        sens.setup_sensitivity(param_list)
+
+        for i, (var, _, _, _) in enumerate(sens.block._sens_data_list):
+            con = sens.block.paramConst[i+1]
+            self.assertEqual(model.sens_state_0[var], i+1)
+            self.assertEqual(model.sens_state_1[var], i+1)
+            self.assertEqual(model.sens_init_constr[con], i+1)
+            self.assertEqual(model.dcdp[con], i+1)
+
+        self.assertTrue(type(model.sens_sol_state_1_z_L) is Suffix)
+        self.assertTrue(type(model.sens_sol_state_1_z_U) is Suffix)
+        self.assertTrue(type(model.ipopt_zL_out) is Suffix)
+        self.assertTrue(type(model.ipopt_zU_out) is Suffix)
+        self.assertTrue(type(model.ipopt_zL_in) is Suffix)
+        self.assertTrue(type(model.ipopt_zU_in) is Suffix)
+        self.assertTrue(type(model.dual) is Suffix)
+        self.assertTrue(type(model.DeltaP) is Suffix)
```

Comment:
Instead of `assertTrue`, use `assertIs`; e.g., `assertIs(type(model.DeltaP), Suffix)`